Test failure is genuine here:
```console
FAILED tests/pylint/test_enforce_type_hints.py::test_invalid_device_class - AssertionError: Expected messages did not match actual.

Expected:
MessageTest(msg_id='hass-return-type', line=12, node=<FunctionDef.device_class l.12 at 0x7f0ad1164550>, args=(['CoverDeviceClass', 'str', None], 'device_class'), confidence=Confidence(name='UNDEFINED', description='Warning without any associated confidence level.'), col_offset=4, end_line=12, end_col_offset=20)

Got:
MessageTest(msg_id='hass-return-type', line=12, node=<FunctionDef.device_class l.12 at 0x7f0ad1164550>, args=(['CoverDeviceClass', None], 'device_class'), confidence=Confidence(name='UNDEFINED', description='Warning without any associated confidence level.'), col_offset=4, end_line=12, end_col_offset=20)
```

--- a/tests/pylint/test_enforce_type_hints.py
+++ b/tests/pylint/test_enforce_type_hints.py
@@ -898,7 +898,7 @@ def test_invalid_device_class(
         pylint.testutils.MessageTest(
             msg_id="hass-return-type",
             node=prop_node,
-            args=(["CoverDeviceClass", "str", None], "device_class"),
+            args=(["CoverDeviceClass", None], "device_class"),
             line=12,
             col_offset=4,
             end_line=12,

